### ISSUE TITLE
LORA训练，loss不降

### ISSUE THREAD

**Issue by gulegeji** [baichuan-inc/Baichuan-13B#47]
使用自己的数据和公开数据，loss都不降低

**Comment by delltower** [comment]
> 使用自己的数据和公开数据，loss都不降低

请贴一下训练参数和框架

**Comment by gulegeji** [comment]
13B LORA，loss一直在3点几，训练了3epoch，框架是https://github.com/hiyouga/LLaMA-Efficient-Tuning/

参数
accelerate launch --config_file configs/accelerate_config src/train_sft.py \
    --model_name_or_path ${MODEL_PATH}\
    --do_train \
    --dataset alpaca_gpt4_zh \
    --lora_target W_pack \
    --finetuning_type lora \
    --output_dir ${OUTPUT_CKPT_DIR}\
    --overwrite_cache \
    --per_device_train_batch_size 4 \
    --gradient_accumulation_steps 4 \
    --lr_scheduler_type cosine \
    --logging_steps 10 \
    --save_steps 1000 \
    --learning_rate 5e-5 \
    --num_train_epochs 3.0 \
    --plot_loss \
    --fp16

finetuning_args.json内容
{
  "finetuning_type": "lora",
  "lora_alpha": 32.0,
  "lora_dropout": 0.1,
  "lora_rank": 8,
  "lora_target": [
    "W_pack"
  ],
  "name_module_trainable": "mlp",
  "num_hidden_layers": 32,
  "num_layer_trainable": 3
}

所有参数
{
  "output_dir": "/saved_models/baichuan-13b-base-dataset-only-prompt-baichuan-test/",
  "overwrite_output_dir": false,
  "do_train": true,
  "do_eval": false,
  "do_predict": false,
  "evaluation_strategy": "no",
  "prediction_loss_only": false,
  "per_device_train_batch_size": 4,
  "per_device_eval_batch_size": 8,
  "per_gpu_train_batch_size": null,
  "per_gpu_eval_batch_size": null,
  "gradient_accumulation_steps": 4,
  "eval_accumulation_steps": null,
  "eval_delay": 0,
  "learning_rate": 5e-05,
  "weight_decay": 0.0,
  "adam_beta1": 0.9,
  "adam_beta2": 0.999,
  "adam_epsilon": 1e-08,
  "max_grad_norm": 1.0,
  "num_train_epochs": 3.0,
  "max_steps": -1,
  "lr_scheduler_type": "cosine",
  "warmup_ratio": 0.0,
  "warmup_steps": 0,
  "log_level": "passive",
  "log_level_replica": "warning",
  "log_on_each_node": true,
  "logging_dir": "/saved_models/baichuan-13b-base-dataset-only-prompt-baichuan-test/runs/Jul13_02-52-59_qgDQjX",
  "logging_strategy": "steps",
  "logging_first_step": false,
  "logging_steps": 10,
  "logging_nan_inf_filter": true,
  "save_strategy": "steps",
  "save_steps": 1000,
  "save_total_limit": null,
  "save_safetensors": false,
  "save_on_each_node": false,
  "no_cuda": false,
  "use_mps_device": false,
  "seed": 42,
  "data_seed": null,
  "jit_mode_eval": false,
  "use_ipex": false,
  "bf16": false,
  "fp16": true,
  "fp16_opt_level": "O1",
  "half_precision_backend": "cuda_amp",
  "bf16_full_eval": false,
  "fp16_full_eval": false,
  "tf32": null,
  "local_rank": 0,
  "ddp_backend": null,
  "tpu_num_cores": null,
  "tpu_metrics_debug": false,
  "debug": [],
  "dataloader_drop_last": false,
  "eval_steps": null,
  "dataloader_num_workers": 0,
  "past_index": -1,
  "run_name": "/saved_models/baichuan-13b-base-dataset-only-prompt-baichuan-test/",
  "disable_tqdm": false,
  "remove_unused_columns": true,
  "label_names": null,
  "load_best_model_at_end": false,
  "metric_for_best_model": null,
  "greater_is_better": null,
  "ignore_data_skip": false,
  "sharded_ddp": [],
  "fsdp": [],
  "fsdp_min_num_params": 0,
  "fsdp_config": {
    "fsdp_min_num_params": 0,
    "xla": false,
    "xla_fsdp_grad_ckpt": false
  },
  "fsdp_transformer_layer_cls_to_wrap": null,
  "deepspeed": null,
  "label_smoothing_factor": 0.0,
  "optim": "adamw_torch",
  "optim_args": null,
  "adafactor": false,
  "group_by_length": false,
  "length_column_name": "length",
  "report_to": [
    "mlflow"
  ],
  "ddp_find_unused_parameters": false,
  "ddp_bucket_cap_mb": null,
  "dataloader_pin_memory": true,
  "skip_memory_metrics": true,
  "use_legacy_prediction_loop": false,
  "push_to_hub": false,
  "resume_from_checkpoint": null,
  "hub_model_id": null,
  "hub_strategy": "every_save",
  "hub_token": "<HUB_TOKEN>",
  "hub_private_repo": false,
  "gradient_checkpointing": false,
  "include_inputs_for_metrics": false,
  "fp16_backend": "auto",
  "push_to_hub_model_id": null,
  "push_to_hub_organization": null,
  "push_to_hub_token": "<PUSH_TO_HUB_TOKEN>",
  "mp_parameters": "",
  "auto_find_batch_size": false,
  "full_determinism": false,
  "torchdynamo": null,
  "ray_scope": "last",
  "ddp_timeout": 1800,
  "torch_compile": false,
  "torch_compile_backend": null,
  "torch_compile_mode": null,
  "xpu_backend": null,
  "sortish_sampler": false,
  "predict_with_generate": false,
  "generation_max_length": 512,
  "generation_num_beams": null,
  "generation_config": null
}

**Comment by gulegeji** [comment]
accelerate参数
compute_environment: LOCAL_MACHINE
deepspeed_config:
  gradient_accumulation_steps: 4
  gradient_clipping: 0.5
  offload_optimizer_device: none
  offload_param_device: none
  zero3_init_flag: false
  zero_stage: 2
distributed_type: DEEPSPEED
downcast_bf16: 'no'
machine_rank: 0
main_training_function: main
mixed_precision: fp16
num_machines: 1
num_processes: 3
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false

**Comment by hiyouga** [comment]
该问题请修复，请阅读项目的 Readme 文件
